File: c++/1092.cpp
#include <algorithm>
#include <iostream>
#include <vector>

using namespace std;

vector<int> crains;
vector<int> boxes;

int main() {
  freopen("example.txt", "r", stdin);

  int N, M;
  cin >> N;
  for (int i = 0; i < N; i++) {
    int a;
    cin >> a;
    crains.push_back(a);
  }

  cin >> M;
  for (int i = 0; i < M; i++) {
    int a;
    cin >> a;
    boxes.push_back(a);
  }

  sort(crains.begin(), crains.end(), greater<int>());
  sort(boxes.begin(), boxes.end(), greater<int>());

  if (crains[0] < boxes[0]) {
    cout << -1;
    return 0;
  }

  int t = 0;

  while (boxes.size()) {
    int cIdx = 0;
    int bIdx = 0;
    while (cIdx < crains.size() && bIdx < boxes.size()) {
      if (boxes[bIdx] <= crains[cIdx]) {
        boxes.erase(boxes.begin() + bIdx);
        cIdx++;
      } else {
        bIdx++;
      }
    }

    t++;
  }

  cout << t;

  return 0;
}

File: js/1092.js
// my solution

// const fs = require('fs');

// const inputs = fs.readFileSync('example.txt').toString().trim().split('\n');

// const N = +inputs[0];
// const crains = inputs[1]
//   .split(' ')
//   .map(Number)
//   .sort((a, b) => b - a);
// let M = +inputs[2];
// const boxes = inputs[3]
//   .split(' ')
//   .map(Number)
//   .sort((a, b) => b - a);

// const isVisited = Array(M).fill(false);

// if (boxes[0] > crains[0]) {
//   console.log(-1);
// } else {
//   let t = 0;
//   while (0 < M) {
//     for (let i = 0; i < crains.length; i++) {
//       for (let j = 0; j < boxes.length; j++) {
//         if (isVisited[j]) continue;
//         if (crains[i] >= boxes[j]) {
//           isVisited[j] = true;
//           M--;
//           break;
//         }
//       }
//     }

//     t++;
//   }

//   console.log(t);
// }

const fs = require('fs');

const inputs = fs.readFileSync('example.txt').toString().trim().split('\n');

const N = +inputs[0];
const crains = inputs[1]
  .split(' ')
  .map(Number)
  .sort((a, b) => b - a);
let M = +inputs[2];
const boxes = inputs[3]
  .split(' ')
  .map(Number)
  .sort((a, b) => b - a);

if (boxes[0] > crains[0]) {
  console.log(-1);
} else {
  let t = 0;
  while (boxes.length) {
    cIdx = 0;
    bIdx = 0;
    while (cIdx < crains.length && bIdx < boxes.length) {
      if (crains[cIdx] >= boxes[bIdx]) {
        cIdx++;
        boxes.splice(bIdx, 1);
      } else {
        bIdx++;
      }
    }

    t++;
  }

  console.log(t);
}
